sbt-dumpschema-plugin/src/main/scala/org/virtuslab/akkasaferserializer/DumpSchemaPluginComponent.scala

```diff
@@ -0,0 +1,76 @@
+package org.virtuslab.akkasaferserializer
+import org.virtuslab.akkasaferserializer.model.{ClassAnnotation, Field, TypeDefinition}
+
+import scala.tools.nsc.{Global, Phase}
+import scala.tools.nsc.plugins.PluginComponent
+
+class DumpSchemaPluginComponent(val options: DumpSchemaOptions, val global: Global) extends PluginComponent {
+  import global._
+  override val phaseName: String = "dump-schema"
+  override val runsAfter: List[String] = List("typer")
+
+  lazy val writer = new SchemaWriter(options)
+
+  override def newPhase(prev: Phase): Phase =
+    new StdPhase(prev) {
+
+      override def apply(unit: global.CompilationUnit): Unit = {
+        val body = unit.body
+        val reporter = CrossVersionReporter(global)
+        val effectNames = Seq("akka.persistence.typed.scaladsl.Effect", "akka.persistence.typed.scaladsl.ReplyEffect")
+
+        def extractFromTypesSealed(tpe: Type): List[Type] = {
```

Comment:
Hmmm actually... 🐘 in the room, the current impl doesn't include classes transitively, right? 🤔

Reply:
E.g. I can see `LifecycleChangeReason` is included as `typeName` in a few places in the current Hydra dump, but the definition of `LifecycleChangeReason` itself is missing... so if the possible values of this enum change, we're not gonna detect that unless classes are included transitively :/

Reply:
On what criteria do we include a class? Blacklist for `scala._` and `akka._` are obvious, but what about 3rd party library (and criteria what is 3rd party library)?

Reply:
Also, what if we traverse into java code? Some data would have already be erased by the time our plugin got there.

Reply:
Let's start with the simplest approach (so probably: include everything, even 3rd party libs?) and see how it works for Phoenix ;)

Reply:
Hmmm if data is lost due to erasure... 🤷🏻 we'll never achieve 100% accurate representation anyway